CMakeLists.txt

```diff
@@ -1,36 +1,59 @@
 # Copyright 2010, Olivier Stasse, JRL, CNRS/AIST
-
 CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
 
-INCLUDE(cmake/base.cmake)
-INCLUDE(cmake/boost.cmake)
-INCLUDE(cmake/eigen.cmake)
-INCLUDE(cmake/pthread.cmake)
-
 SET(PROJECT_ORG stack-of-tasks)
 SET(PROJECT_NAME dynamic-graph)
 SET(PROJECT_DESCRIPTION "Dynamic graph library")
 SET(PROJECT_URL "http://github.com/${PROJECT_ORG}/${PROJECT_NAME}")
 SET(PROJECT_SUFFIX "-v3")
 
-SET(CUSTOM_HEADER_DIR "${PROJECT_NAME}")
+# Export CMake Target
+SET(PROJECT_USE_CMAKE_EXPORT TRUE)
+
+# Make sure that every header is generated in dynamic-graph
+SET(CUSTOM_HEADER_DIR ${PROJECT_NAME})
+
+# Do not fail on warning
+SET(CXX_DISABLE_WERROR TRUE)
 
+INCLUDE(cmake/base.cmake)
+
+# Specify the project.
+cmake_policy(SET CMP0048 NEW)
+PROJECT(${PROJECT_NAME}
+  LANGUAGES
+  CXX
+  VERSION
+  ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}
+  )
+
+
+# jrl-cmakemodules handling of usual modules
+INCLUDE(cmake/header.cmake)
+INCLUDE(cmake/pthread.cmake)
+add_project_dependency(Eigen3 REQUIRED NO_MODULE)
+add_project_dependency(Boost REQUIRED
+  OPTIONAL_COMPONENTS
+  serialization system unit_test_framework thread)
+# Allows math in mathjax.
 SET(DOXYGEN_USE_MATHJAX YES)
 
 # Defines paths.
 SET(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
 
-SETUP_PROJECT()
-
 # Add configuration headers for plug-ins.
 GENERATE_CONFIGURATION_HEADER(
   ${HEADER_DIR} config-tracer.hh DG_TRACER tracer_EXPORTS)
 GENERATE_CONFIGURATION_HEADER(
   ${HEADER_DIR}
   config-tracer-real-time.hh DG_TRACERREALTIME tracer_real_time_EXPORTS)
 
+set(PACKAGE_EXTRA_MACROS
+  "set (DYNAMIC_GRAPH_PLUGINDIR \${CMAKE_CURRENT_LIST_DIR}/../../plugin)")
```

Comment:
This `CMAKE_CURRENT_LIST_DIR` will be defined to the directory where this project have been compiled. I don't think this is relevant to export it. Maybe you mean something like `${CMAKE_INSTALL_FULL_LIBDIR}/plugin`.

Reply:
Well not exactly. This is relative to where this file is installed.

Reply:
Ok, I got confused by the "Full directory" in the docs